wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileManager.java

```diff
@@ -82,17 +82,21 @@ public Devfile parse(String devfileContent, boolean verbose)
    * Creates {@link WorkspaceImpl} from given devfile with available name search
    *
    * @param devfile source devfile
+   * @param recipeFileContentProvider content provider for recipe-type tool
    * @return created {@link WorkspaceImpl} instance
+   * @throws DevfileException when general devfile error occurs
```

Comment:
Sorry I don't fully understand what does `when general devfile error occurs` mean.
I think it would be better to rewrite java doc in the following way:
```java
   * @throws DevfileFormatException when devfile integrity validation fail
   * @throws DevfileRecipeFormatException when devfile recipe format is invalid
   * @throws DevfileException when any another devfile related error occurs
```